### ISSUE TITLE
Add option to choose dataset shuffle in dataloders

### ISSUE THREAD

**Issue by icedoom888** [ecmwf/anemoi-core#813]
### Is your feature request related to a problem? Please describe.

Currently the codebase is hardcoded:
training has shuffle= True
validation has shuffle = False
test has shuffle = False

### Describe the solution you'd like

These should be extended to allow more flexibility in experiments:

```
@cached_property
    def ds_train(self) -> MultiDataset:
        """Create multi-dataset for training."""
        return self._get_dataset(
            self.train_dataloader_config,
            shuffle=True,
            label="training",
        )

    @cached_property
    def ds_valid(self) -> MultiDataset:
        """Create multi-dataset for validation."""
        return self._get_dataset(
            self.valid_dataloader_config,
            shuffle=False,
            val_rollout=self.config.dataloader.validation_rollout,
            label="validation",
        )

    @cached_property
    def ds_test(self) -> MultiDataset:
        """Create multi-dataset for testing."""
        return self._get_dataset(self.test_dataloader_config, shuffle=False, label="test")
```

**Comment by icedoom888** [comment]
Hey @mchantry, i had this fixed before the multi dataset PR in another branch but now with the new way the schemas are build it doesn't apply anymore, so decided to open an issue to highlight this, in case someone finds another solution! 

At some point i found myself in the need of not shuffling the training set: wanted to limit batches to 1 and overfit on one sample for some experiments. Found this limitation ;)

**Comment by anaprietonem** [comment]
Hey, while this seems like an easy change we could make in our code to allow extra configurability, I think we should be careful about allowing things just for the sake of testing. Alberto, do you have an example where this has proven useful?

One concern is that it makes it easier for people to make mistakes (e.g., having training data not shuffled). Another is that in validation this could introduce issues, since the plotting code would then see different samples each epoch, making those metrics harder to evaluate.

For these reasons, my proposal would be not to allow this unless there is a clear example of the benefit it brings.
